docs/docs/contributing/scala2-vs-scala3.md

```diff
@@ -0,0 +1,38 @@
+# Divergences between Scala 2 and Dotty
+The following issues encountered when compiling Scala 2 code as-is under Dotty:
+
+## Scalafix candidates
+- If a method is defined `toSet()`, it cannot be called `toSet`.
+- “result type of implicit definition needs to be given explicitly”
+- There are no `'Symbol`s in Scala 3, you must construct symbols via `new Symbol("foo")` instead of old `'foo`
+
+## Trivial
+- Scala 2.13 libraries cannot be used from Dotty, because the type signatures have Scala version `5.3` but `5.0` is expected.
+- To use Scala 2.12 dependencies from SBT with Dotty, explicitly suffix their names with `_2.12`.
```

Comment:
This works but it's not the recommended way to do it, use `withDottyCompat` instead as documented in https://github.com/lampepfl/dotty-example-project#getting-your-project-to-compile-with-dotty

Reply:
Good catch! Didn't know about it. The fact that we need to pass a `scalaVersion.value` to it kind of kills the entire point of `%%` as opposed to `%` though.

Reply:
Using `%` with a fixed suffix makes cross-compilation impossible, whereas using `withDottyCompat(scalaVersion.value)` means the project can cross-compile between multiple versions of Scala 2 and Dotty and will always use the correct suffix.

Reply:
I meant, ideally, we should obtain `scalaVersion` from the plugin itself so that the user doesn't have to pass it explicitly.

Reply:
I couldn't find a way to do that unfortunately, `scalaVersion.value` is an sbt setting which you can't compute from a regular method like `withDottyCompat`.